woocommerce/woocommerce-ios
# World's worst HACK week code – Remote Tap to Pay on iPhone

Closes: Nothing. Never merge this. Not even a bit.



## Description

This is a hack week project. Don't merge it, don't judge it, ideally don't even read it, it's just cool that it works. 
I've used some of [Apple's sample networking code](https://developer.apple.com/documentation/Network/building-a-custom-peer-to-peer-protocol)

## Testing information


Hahahahaha good luck.

1. Launch the app on an iPhone, and sign out if you're not already.
2. Tap `Remote Tap to Pay` on the final Login prologue screen – you might need to tap skip a few times to get to this.
3. Launch the app on an iPad and switch to a POS eligible store
4. Tap `Connect your reader` and choose remote tap to pay
5. Do it again if it doesn't work the first time
6. Add something to your cart and checkout as normal
7. Tap your card on the iPhone
8. Observe that the payment completes
9. Do a little dance
10. Never read the code.
